### PR TITLE
API-4282: created new endpoint for TEAM_MEMBER_ADDED event and saved in repo

### PR DESCRIPTION
**Pull request by mattclark-zerogravit**

### CODE REVIEW

File: app/uk/gov/hmrc/apiplatformevents/repository/ApplicationEventsRepository.scala [hunk 1]
```diff
@@ -17,41 +17,39 @@
...
       ReactiveMongoFormats.objectIdFormats) {
 
-  def createEntity(event: ApiPlatformEventsDBModel)(
-      implicit ec: ExecutionContext): Future[Unit] =
-    insert(event).map(_ => ())
+  implicit val dateFormat = ReactiveMongoFormats.dateTimeFormats
+
+  private[repository] def convertEventAndInsert(event: ApplicationEvent)(implicit executionContext: ExecutionContext)  = {
+    event match {
+      case tmae: TeamMemberAddedEvent => {
```
Comment (lburgos) [inline app/uk/gov/hmrc/apiplatformevents/repository/ApplicationEventsRepository.scala:43]: I don't think we need the pattern matching as we'll just do the same with every event (it's enough knowing they're all ApplicationEvent objects)

File: app/uk/gov/hmrc/apiplatformevents/services/ApplicationEventsService.scala [hunk 2]
```diff
@@ -18,26 +18,26 @@ package uk.gov.hmrc.apiplatformevents.services
...
-      ApiPlatformEventsDBModel(event.parameter1,
-                               event.parameter2,
-                               event.telephoneNumber,
-                               event.emailAddress)
-    )
+      ec: ExecutionContext): Future[Boolean] = {
+    repo.createEntity(event).recover {
+      case NonFatal(e) => {
+        Logger.info("Exception happened when trying to createEntity:",e)
+        false
```
Comment (lburgos) [inline app/uk/gov/hmrc/apiplatformevents/services/ApplicationEventsService.scala:38]: I find it easier to recover at the controller level as it's at that point that you know what kind of response you want to return (so for example, a 404 for a NotFoundException and a 500 for any other exception you wouldn't expect)

Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/apiplatformevents/services/ApplicationEventsService.scala]: will change

File: app/uk/gov/hmrc/apiplatformevents/repository/ApplicationEventsRepository.scala (hunk 1 above)
Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/apiplatformevents/repository/ApplicationEventsRepository.scala]: will refactor this

**Pull request merged**